src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -169,32 +165,21 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
             }
           }
         });
-    }
     return renamedPerms;
   }
 
-  private List<OkapiPermission> getNewAndModifiedPerms(ModuleId moduleId,
-      Map<String, Permission> dbPerms, List<OkapiPermission> okapiPerms) {
-    List<OkapiPermission> newPerms = getNewPerms(dbPerms, okapiPerms);
-    List<OkapiPermission> modifiedPerms = getModifiedPerms(moduleId, dbPerms, okapiPerms);
-    newPerms.addAll(modifiedPerms);
-    return newPerms;
-  }
-
   private List<OkapiPermission> getModifiedPerms(ModuleId moduleId, Map<String, Permission> dbPerms,
-      List<OkapiPermission> okapiPerms) {
+      @NotNull List<OkapiPermission> okapiPerms) {
     List<OkapiPermission> modifiedPerms = new ArrayList<>();
 
-    if(okapiPerms != null) {
-      okapiPerms.stream()
+    okapiPerms.stream()
         .filter(okapiPerm -> {
-            String name = okapiPerm.getPermissionName();
-            return dbPerms.containsKey(name)
-                && (dbPerms.get(name).getDeprecated()
-                || !PermissionUtils.equals(okapiPerm, moduleId.getProduct(), dbPerms.get(name)));
-          })
+          String name = okapiPerm.getPermissionName();
+          return dbPerms.containsKey(name)
+              && (dbPerms.get(name).getDeprecated()
+              || !PermissionUtils.equals(okapiPerm, moduleId.getProduct(), dbPerms.get(name)));
+        })
         .forEach(modifiedPerms::add);
```

Comment:
```
        .collect(Collectors.toList());
```
avoids modifiedPerms and is more stream-like.

Reply:
ok